buildspec/linuxIntegrationTests.yml

```diff
@@ -47,6 +47,7 @@ phases:
         commands:
             - export AWS_TOOLKIT_TEST_USER_DIR=/tmp/
             - export AWS_TOOLKIT_TEST_NO_COLOR=1
+            - export NO_COVERAGE=true
```

Comment:
It would be preferable to avoid accruing technical debt while addressing the issue.

Looking at the test output, it seems that the extension is already activated by the time the test tries to  activate it, which is to be expected based on this PRs change.  [configureToolkitLogging](https://github.com/aws/aws-toolkit-vscode/blob/kantc/startupactivate/src/integrationTest/sam.test.ts#L142) is indicating that the logger is not what is expected. Could we write out what the log type is? Knowing if it is `undefined`, or a `TestLogger`, or something else altogether would help get to the core issue.

Its plausible that the way the sam integration tests work with the logger need some revisions based on the new activation behavior, which makes sense to deal with that when making the activation behavior changes.

Reply:
>  configureToolkitLogging is indicating that the logger is not what is expected. Could we write out what the log type is? Knowing if it is undefined, or a TestLogger, or something else altogether would help get to the core issue.

It's `TestLogger`. None of the global context is set when the extension activates itself and coverage is run. So it can only see the test context which is set by `globalSetup` and another test.

The sam test expects to have access to the same state set by the extension. It reaches into the modules with global variables like `samCliContext` and `logger` and expects to see the ones set during `activate`. When the extension activates itself and the coverage runs, this is absent.

It likely has something to do with the augmentation that the coverage does. Perhaps the extension activates using the files before coverage and the tests run using the files after coverage. So they're actually looking at two different sets of files.

> It would be preferable to avoid accruing technical debt while addressing the issue.

The coverage (and augmentation process) itself needs a lot of work to be made functional. Right now it messes with the source files in unpredictable ways. It's quite broken in how it checks for test coverage at the moment. I've seen it mark completely tested files as "un-hit" during coverage reporting. Additionally, coverage reports for integration tests are  unused right now when they should ideally be merged with the total coverage report for the code base.

tl;dr 
The coverage has a lot of issues that need to be addressed. Will create a follow-up item for a well-needed deep dive and overhaul so we can get an accurate report of the state of the codebase to help improve and maintain code quality.